### ISSUE TITLE
Video sometimes blackscreens instead of showing viz, even though data is coming through fine.

### ISSUE THREAD

**Issue by caass** [datafruits/datafruits#496]
[Original discussion](https://discordapp.com/channels/192542105741033472/739697514860642415/746186781770252319).

There seems to be an issue with videojs and parsing JSON data. First I'll show the stack traces from chrome and firefox.

Chrome:
```
chunk.9862f2cc349c4819ee3b.js:190 VIDEOJS: ERROR: SyntaxError: Unexpected token ' in JSON at position 1
    at JSON.parse (<anonymous>)
    at e.exports (chunk.9862f2cc349c4819ee3b.js:2530)
    at Function.t.getTagSettings (chunk.9862f2cc349c4819ee3b.js:1255)
    at new t (chunk.9862f2cc349c4819ee3b.js:1083)
    [...]
```

Firefox:
```
VIDEOJS: ERROR: SyntaxError: JSON.parse: expected property name or '}' at line 1 column 2 of the JSON data
    exports https://datafruits.fm/assets/chunk.9862f2cc349c4819ee3b.js:2530
    getTagSettings https://datafruits.fm/assets/chunk.9862f2cc349c4819ee3b.js:1255
    t https://datafruits.fm/assets/chunk.9862f2cc349c4819ee3b.js:1083
    [...]
```

Diving into the devtools, let's first take a look at `getTagSettings`. The source code is [here](https://github.com/videojs/video.js/blob/master/src/js/player.js#L4668-L4716). The important parts are here:

```js
static getTagSettings(tag) {
    const baseOptions = {
      sources: [],
      tracks: []
    };

    const tagOptions = Dom.getAttributes(tag);
    const dataSetup = tagOptions['data-setup'];

  // ...

  if (dataSetup !== null) {
      // Parse options JSON
      // If empty string, make it a parsable json object.
      const [err, data] = safeParseTuple(dataSetup || '{}');  // <--- error happens here

      assign(tagOptions, data);
    }

      if (err) {
        log.error(err);  // <--- error is logged here
      }

    assign(baseOptions, tagOptions);
```

So what's happening is that there *is* some value for `dataSetup`, it's just not parseable. So looking up the call stack one time, let's see what's getting passed as the `tag` parameter.

```html
<div muted="true" playsinline="true" data-setup="{'fluid': true}" preload="auto" class="video-js vjs-default-skin vjs-16-9 vjs-big-play-centered vjs-controls-disabled vjs-workinghover vjs-v7 vjs-user-inactive vjs-playing vjs-has-started vjs-live video-player-dimensions" id="video-player" tabindex="-1" role="region" aria-label="Video Player" lang="en-us"><video id="video-player_html5_api" class="vjs-tech" preload="auto" data-setup="{'fluid': true}" playsinline="playsinline" muted="muted" tabindex="-1" src="blob:https://datafruits.fm/1564bc99-4483-7b48-bde6-92bf309c5a4f">
  <!-- Stuff -->
</div>
```

The issue is that `data-setup="{'fluid': true}"` uses single quotes, which is [not allowed in JSON](https://stackoverflow.com/a/36038497).

In conclusion: I hate computers, and the Industrial Revolution and its consequences have been a disaster for the human race.
